wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+        """
+
+        # This is intentionally empty, because we are using the combination
+        # of wx.BufferedPaintDC + an empty OnEraseBackground event to
+        # reduce flicker.
+        pass
+
+    def OnMouseClick(self, event):
+        """
+        Handles the ``wx.EVT_LEFT_DOWN`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.MouseEvent` to be processed.
+        :type `event`: `wx.MouseEvent`
+        """
+
+        if not self.IsEnabled():
+            # Nothing to do, we are disabled.
+            return
+
+        self.SendCheckBoxEvent()
+        event.Skip()
+
+    def SendCheckBoxEvent(self):
+        """ Actually sends the wx.wxEVT_COMMAND_CHECKBOX_CLICKED event. """
+
+        # This part of the code may be reduced to a 3-liner code
+        # but it is kept for better understanding the event handling.
+        # If you can, however, avoid code duplication; in this case,
+        # I could have done:
+        #
+        # self._checked = not self.IsChecked()
+        # checkEvent = wx.CommandEvent(wx.wxEVT_COMMAND_CHECKBOX_CLICKED,
+        #                              self.GetId())
+        # checkEvent.SetInt(int(self._checked))
+        if self.IsChecked():
+
+            # We were checked, so we should become unchecked.
+            self._checked = False
+
+            # Fire a wx.CommandEvent: this generates a
+            # wx.wxEVT_COMMAND_CHECKBOX_CLICKED event that can be caught by the
+            # developer by doing something like:
+            # MyCheckBox.Bind(wx.EVT_CHECKBOX, self.OnCheckBox)
+            checkEvent = wx.CommandEvent(wx.wxEVT_COMMAND_CHECKBOX_CLICKED,
+                                         self.GetId())
+
+            # Set the integer event value to 0 (we are switching to unchecked state).
+            checkEvent.SetInt(0)
+
+        else:
+
+            # We were unchecked, so we should become checked.
+            self._checked = True
+
+            checkEvent = wx.CommandEvent(wx.wxEVT_COMMAND_CHECKBOX_CLICKED,
+                                         self.GetId())
+
+            # Set the integer event value to 1 (we are switching to checked state).
+            checkEvent.SetInt(1)
+
+        # Set the originating object for the event (ourselves).
+        checkEvent.SetEventObject(self)
+
+        # Watch for a possible listener of this event that will catch it and
+        # eventually process it.
+        self.GetEventHandler().ProcessEvent(checkEvent)
+
+        # Refresh ourselves: the bitmap has changed.
+        self.Refresh()
+
+
+# -----------------------------------------------------------------------------
+
+
+class DefineNativeCheckBoxBitmapsMixin():
+    """
+    Inherit in your :class:`wx.Window` based subclass and call
+    ``self.DefineNativeCheckBoxBitmaps()`` in your __init__ def or startup routine
+    to define the native CheckBox Bitmaps as attributes the user can customize other
+    widgets appearance with.
```

Comment:
This sentence is not easy to parse. Perhaps separate it into what-is-this-class and how-you-use-it, and maybe a simple example.

Reply:
Added basic gist of how I started MyCheckListBoxSTC as a sample example and made wording a bit clearer.
Also added current and pressed to the defined bitmaps.